dandischema/models.py

```diff
@@ -200,6 +200,8 @@ def schema_extra(schema: Dict[str, Any], model) -> None:
                     value["title"] = name2title(prop)
                 if value.get("format", None) == "uri":
                     value["maxLength"] = 1000
+                if value.get("pattern", "").startswith("^http"):
+                    value["format"] = "uri"
```

Comment:
shouldn't then this come before preceding `if value.get("format", None) == "uri":` so it gets maxLength set too?